### PR TITLE
Fix parsing of sample import response

### PR DESCRIPTION
**Pull request by Westwooo**

### CODE REVIEW

File: .github/bin/init-cluster [hunk 1]
```diff
@@ -31,7 +31,7 @@ options = {
   password: ENV.fetch("CB_PASSWORD", "password"),
   bucket: ENV.fetch("CB_BUCKET", "default"),
   sec_bucket: ENV.fetch("CB_SEC_BUCKET", ""),
-  server_quota: ENV.fetch("CB_SERVER_QUOTA", 2048).to_i,
```
Comment (chvck) [inline .github/bin/init-cluster:34]: Why this change? We should probably also move to using dinocluster instead of this but that's a different topic...

File: src/cli/buckets_sample.rs [hunk 2]
```diff
@@ -103,32 +86,41 @@ fn load_sample_bucket(
...
+                collected.add_string("status", "success", span);
+            }
             _ => {
-                return Err(unexpected_status_code_error(
-                    response.status(),
-                    response.content(),
-                    span,
-                ))
+                // The format of the returned error varies between server versions so we have to do this substring matching
+                let error = if response.content().contains("already loaded") {
```
Comment (chvck) [inline src/cli/buckets_sample.rs:106]: Typically I'd say that string matching is going to lead to pain down the road. Repeatedly we've seen that the server changes the message and error handling gets broken on clients. Admittedly there's not really a better way but can we just continue surfacing the error message from the server directly?

File: tests/buckets_sample.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+mod common;
```
Comment (chvck) [inline tests/buckets_sample.rs:1]: Nice!

File: .github/bin/init-cluster (hunk 1 above)
Comment (Westwooo) [reply .github/bin/init-cluster:34]: I need to hit the RAM quota limit to test the error that we get when trying to import a sample bucket with less that 200MB left. But I think I'm going to leave the quota and create some temp bucket to take up the space required.

File: tests/query.rs [hunk 4]
```diff
@@ -139,10 +139,10 @@ fn fetch_meta() {
                 playground::RetryExpectations::ExpectOut,
                 |json| -> TestResult<bool> {
                     val = json.clone();
-                    Ok(true)
+                    // Wait until the document has been indexed
+                    Ok(val["results"].as_array().unwrap().len() == 1)
```
Comment (Westwooo) [inline tests/query.rs:143]: This test was flaky due to a race condition where the index would be created but the document created in the test had not been indexed yet, so the query would succeed but there would be no results.

File: src/cli/buckets_sample.rs [hunk 5]
```diff
@@ -103,32 +86,44 @@ fn load_sample_bucket(
...
+            }
             _ => {
-                return Err(unexpected_status_code_error(
-                    response.status(),
-                    response.content(),
-                    span,
-                ))
+                // Errors are in the form: `["error msg here"]` so need to drop the first and last 2 chars
+                let mut chars = response.content().chars();
+                chars.next();
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply src/cli/buckets_sample.rs]: We can, done!

File: src/cli/buckets_sample.rs [hunk 6]
```diff
@@ -103,32 +86,39 @@ fn load_sample_bucket(
...
             _ => {
-                return Err(unexpected_status_code_error(
-                    response.status(),
-                    response.content(),
-                    span,
-                ))
+                let msg = if response.content().contains("\"errors\":") {
+                    // In 7.6 the content is in the form '"{"errors":["Err msg here"]}"' so we need to remove the prepended
+                    // "errors:" and the extra trailing char
+                    &response.content()[12..response.content().len() - 2]
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply src/cli/buckets_sample.rs:108]: That's why I went with the string matching approach originally, since trying to trim the message from the server just opens a whole can of worms

**Review by chvck — APPROVED** [review]
